alien4cloud/utils.go

```diff
@@ -74,3 +74,13 @@ func (jar *jar) SetCookies(u *url.URL, cookies []*http.Cookie) {
 func (jar *jar) Cookies(u *url.URL) []*http.Cookie {
 	return jar.cookies[u.Host]
 }
+
+func readBodyData(response *http.Response, data interface{}) error {
+	defer response.Body.Close()
```

Comment:
Callers of readBodyData() are already taking care of defering the close response.Body.Close().
Maybe this coudl be removed here, to have the code creating the response taking care of closing it

Reply:
Yeah good point, actually I was wondering if it worthwhile to do the opposite closing here and do not do it anywhere because the doc says that the body should always be read to EOF in order to reuse connection. (https://pkg.go.dev/net/http?tab=doc#Client.Do)

> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close. If the Body is not both read to EOF and closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.

So I was thinking about a refactoring to take care of this.

Reply:
Ok, then maybe the function could be called something like readDataCloseResponseBody()